datashader/transfer_functions.py

```diff
@@ -97,12 +109,22 @@ def interpolate(agg, low="lightblue", high="darkblue", how='cbrt'):
     how = _normalize_interpolate_how(how)
     data = how(agg - offset)
     span = [np.nanmin(data), np.nanmax(data)]
-    rspan, gspan, bspan = zip(rgb(low), rgb(high))
-    r = np.interp(data, span, rspan, left=255).astype(np.uint8)
-    g = np.interp(data, span, gspan, left=255).astype(np.uint8)
-    b = np.interp(data, span, bspan, left=255).astype(np.uint8)
-    a = np.where(np.isnan(data), 0, 255).astype(np.uint8)
-    img = np.dstack([r, g, b, a]).view(np.uint32).reshape(a.shape)
+    if isinstance(cmap, list):
+        rspan, gspan, bspan = np.array(list(zip(*map(rgb, cmap))))
+        span = np.linspace(span[0], span[1], len(cmap))
+        r = np.interp(data, span, rspan, left=255).astype(np.uint8)
+        g = np.interp(data, span, gspan, left=255).astype(np.uint8)
+        b = np.interp(data, span, bspan, left=255).astype(np.uint8)
+        a = np.where(np.isnan(data), 0, 255).astype(np.uint8)
+        img = np.dstack([r, g, b, a]).view(np.uint32).reshape(a.shape)
+    elif (getattr(type(cmap), '__module__', '').startswith('matplotlib') and
+          callable(cmap)):
```

Comment:
I'm assuming that you're not doing `issubclass(matplotlib.colors.Colormap)` to avoid importing matplotlib?  If so, why not just check that it's callable and try calling it, expecting some error to be raised if it's not something that works like a Matplotlib Colormap?  The current approach won't allow people to subclass that class or to provide duck-typed workalike objects, and I'm not sure what we gain from disallowing that.  I'd like to avoid mucking around with the module names if we don't have to...